kvbc/include/sparse_merkle/histograms.h

```diff
@@ -0,0 +1,129 @@
+
+  // Used in merkle_tree_db_adapter.cpp
+  std::shared_ptr<Recorder> dba_batch_to_db_updates = std::make_shared<Recorder>(1, MAX_NS * 5, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_get_value = std::make_shared<Recorder>(1, MAX_NS * 5, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_get_genesis_block_id = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_get_last_reachable_block_id =
+      std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_get_latest_block_id = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_create_block_node = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_get_raw_block = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_last_reachable_block_db_updates =
+      std::make_shared<Recorder>(1, MAX_NS * 5, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_size_of_updates = std::make_shared<Recorder>(1, MAX_VAL_SIZE, 3, Unit::BYTES);
+  std::shared_ptr<Recorder> dba_get_internal = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_serialize_internal = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_serialize_leaf = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_deserialize_internal = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_deserialize_block = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_deserialize_leaf = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_link_st_chain = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_num_blocks_for_st_link = std::make_shared<Recorder>(1, 1000000, 3, Unit::COUNT);
+  std::shared_ptr<Recorder> dba_add_raw_block = std::make_shared<Recorder>(1, MAX_NS * 10, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_delete_block = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_get_leaf_key_val_at_most_version =
+      std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_delete_keys_for_block = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_has_block = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_keys_for_version = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_hash_parent_block = std::make_shared<Recorder>(1, MAX_NS, 3, Unit::NANOSECONDS);
+  std::shared_ptr<Recorder> dba_hashed_parent_block_size = std::make_shared<Recorder>(1, MAX_VAL_SIZE, 3, Unit::BYTES);
+};
+
+static Recorders histograms;
```

Comment:
If I am not missing something, I think there is a problem with using a `static` variable in a header in this context. What I mean is that we will have multiple `Recorders` instances in different translation units. However, the constructors of these instances will add to a common `singleton`, throwing exceptions for existing components in `PerformanceHandler::registerComponent()`.

I think a simple solution would be to declare the variable as `inline const Recorders histograms;`.

Reply:
Fixed.